libraries/botbuilder-core/src/cloudAdapterBase.ts

```diff
@@ -0,0 +1,356 @@
+                    context.turnState.set(INVOKE_RESPONSE_KEY, activity);
+                } else if (activity.type === ActivityTypes.Trace && activity.channelId !== Channels.Emulator) {
+                    // no-op
+                } else {
+                    const connectorClient = context.turnState.get<ConnectorClient>(this.ConnectorClientKey);
+                    if (!connectorClient) {
+                        throw new Error('Unable to extract ConnectorClient from turn context.');
+                    }
+
+                    if (activity.replyToId) {
+                        return connectorClient.conversations.replyToActivity(
+                            activity.conversation.id,
+                            activity.replyToId,
+                            activity
+                        );
+                    } else {
+                        return connectorClient.conversations.sendToConversation(activity.conversation.id, activity);
+                    }
+                }
+
+                return { id: activity.id ?? '' };
+            })
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async updateActivity(context: TurnContext, activity: Partial<Activity>): Promise<ResourceResponse | void> {
+        if (!context) {
+            throw new TypeError('`context` parameter required');
+        }
+
+        if (!activity) {
+            throw new TypeError('`activity` parameter required');
+        }
+
+        const connectorClient = context.turnState.get<ConnectorClient>(this.ConnectorClientKey);
+        if (!connectorClient) {
+            throw new Error('Unable to extract ConnectorClient from turn context.');
+        }
+
+        const response = await connectorClient.conversations.updateActivity(
+            activity.conversation.id,
+            activity.id,
+            activity
+        );
+
+        return response?.id ? { id: response.id } : undefined;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async deleteActivity(context: TurnContext, reference: Partial<ConversationReference>): Promise<void> {
+        if (!context) {
+            throw new TypeError('`context` parameter required');
+        }
+
+        if (!reference) {
+            throw new TypeError('`reference` parameter required');
+        }
+
+        const connectorClient = context.turnState.get<ConnectorClient>(this.ConnectorClientKey);
+        if (!connectorClient) {
+            throw new Error('Unable to extract ConnectorClient from turn context.');
+        }
+
+        await connectorClient.conversations.deleteActivity(reference.conversation.id, reference.activityId);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async continueConversation(
+        reference: Partial<ConversationReference>,
+        logic: (context: TurnContext) => Promise<void>
+    ): Promise<void> {
+        return this.processProactive(
+            this.createClaimsIdentity(),
```

Comment:
@joshgummersall is looking into revising the `BotAdapter.continueConversation()` scenario in JS and changing it to a non-abstract method, more closely aligned with the .NET implementation. 

This would remove the `continueConversationWithClaims()` method added in this PR and turn it into an overload of `continueConversation()`. Additionally, an overload with the appId would be added which would create a valid ServiceClientCredentials for use by the ConnectorClient and UserTokenClient.

```ts
public async continueConversation(
    appId: string,
    reference: Partial<ConversationReference>,
    logic: (context: TurnContext) => Promise<void>
): Promise<void>;
```

Reply:
Please and thank you!

Reply:
Addressed via adding `continueConversationAsync`